src/strpkg/unpack.nim

```diff
@@ -0,0 +1,90 @@
+import streams
+import msgpack4nim
+export msgpack4nim
+import strformat
+import hts/bam
+import strutils
+
+import ./extract
+
+proc `==`(a, b:Target): bool =
+  return a.tid == b.tid and a.length == b.length and a.name == b.name
+
+proc same*(a:seq[Target], b:seq[Target]): bool =
+  if a.len != b.len:
+    return false
+  for i, aa in a:
+    if aa != b[i]:
+      return false
+  return true
+
+proc unpack_type[ByteStream](s: ByteStream, x: var tread) =
+  s.unpack(x.tid)
+  s.unpack(x.position)
+  s.unpack(x.repeat)
+  var f:uint16
+  s.unpack(f)
+  x.flag = Flag(f)
+  var split:uint8
+  s.unpack(split)
+  x.split = Soft(split)
+  s.unpack(x.mapping_quality)
+  s.unpack(x.repeat_count)
+  s.unpack(x.align_length)
+  var L:uint32 = 0
+  var qname: string
+  s.unpack(L)
+  qname = newString(L)
+  if L > 0'u32:
+    s.unpack(qname)
+  x.qname = qname
+
+
+proc unpack_file*(fs:FileStream, expected_format_version:int16=0): tuple[targets: seq[Target], fragment_distribution: array[4096, uint32], reads: seq[tread]] =
+
+  var str = fs.readStr(3)
+  doAssert str == "STR", "[strling] expected bin file to start with \"STR\""
```

Comment:
we should probably add an error message indicating that they need to re-run the extract step with the new version